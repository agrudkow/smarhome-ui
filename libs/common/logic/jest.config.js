module.exports = {
  name: 'common-logic',
  preset: '../../../jest.config.js',
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'html'],
  coverageDirectory: '../../../coverage/libs/common/logic',
};
