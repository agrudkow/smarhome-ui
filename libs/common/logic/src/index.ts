export * from './lib/use-accordion';
export * from './lib/use-window-dimensions';
