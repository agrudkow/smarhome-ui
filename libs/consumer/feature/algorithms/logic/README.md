# consumer-feature-algorithms-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-algorithms-logic` to execute the unit tests via [Jest](https://jestjs.io).
