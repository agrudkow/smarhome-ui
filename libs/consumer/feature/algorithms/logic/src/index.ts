export * from './lib/algorithm.interface';
export * from './lib/algorithms-cells-parser';
export * from './lib/algorithms-data-parser';
