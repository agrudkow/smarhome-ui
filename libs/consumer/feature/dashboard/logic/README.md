# consumer-feature-dashboard-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-dashboard-logic` to execute the unit tests via [Jest](https://jestjs.io).
