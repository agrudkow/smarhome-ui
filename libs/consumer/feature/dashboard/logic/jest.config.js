module.exports = {
  name: 'consumer-feature-dashboard-logic',
  preset: '../../../../../jest.config.js',
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'html'],
  coverageDirectory:
    '../../../../../coverage/libs/consumer/feature/dashboard/logic',
};
