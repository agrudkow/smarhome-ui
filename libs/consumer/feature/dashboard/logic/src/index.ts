export * from './lib/consumer-feature-dashboard-logic';
