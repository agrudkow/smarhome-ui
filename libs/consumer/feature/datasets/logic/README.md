# consumer-feature-datasets-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-datasets-logic` to execute the unit tests via [Jest](https://jestjs.io).
