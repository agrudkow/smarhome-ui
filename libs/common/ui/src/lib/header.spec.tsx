import React from 'react';
import Header from './header';
import { customRender } from '@smarthome/common/logic';

describe(' Header', () => {
  it('should render successfully', () => {
    const { baseElement } = customRender(
      <Header
        showViewName={false}
        viewName="Test"
        onButtonClick={jest.fn()}
        loading={false}
      />
    );
    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot', () => {
    const { baseElement } = customRender(
      <Header
        showViewName={false}
        viewName="Test"
        onButtonClick={jest.fn()}
        loading={false}
      />
    );
    expect(baseElement.firstChild).toMatchSnapshot();
  });

  it('should set view name', () => {
    const viewName = 'Test view name';
    const { getByText } = customRender(
      <Header
        showViewName={false}
        viewName={viewName}
        onButtonClick={jest.fn()}
        loading={false}
      />
    );
    expect(getByText(viewName)).toBeTruthy();
  });
});
