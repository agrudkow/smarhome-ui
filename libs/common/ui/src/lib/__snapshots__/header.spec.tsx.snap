// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Header should match snapshot 1`] = `
.c0 {
  position: -webkit-sticky;
  position: -webkit-sticky;
  position: sticky;
  z-index: 9999;
  width: 100%;
  height: 40px;
  top: 0;
  left: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  overflow: hidden;
}

.c1 {
  background-color: #102128;
  width: 100%;
  height: 36px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  overflow: hidden;
}

.c2 {
  color: #555;
  font-size: 14px;
  line-height: 20px;
  -webkit-letter-spacing: -0.3px;
  -moz-letter-spacing: -0.3px;
  -ms-letter-spacing: -0.3px;
  letter-spacing: -0.3px;
  color: #E3E2E1;
  white-space: nowrap;
  margin: 0 15px;
}

.c3 {
  color: #555;
  font-size: 14px;
  line-height: 20px;
  -webkit-letter-spacing: -0.3px;
  -moz-letter-spacing: -0.3px;
  -ms-letter-spacing: -0.3px;
  letter-spacing: -0.3px;
  color: #E3E2E1;
  white-space: nowrap;
  margin: 0 15px;
}

.c4 {
  width: 100%;
  background-color: #102128;
  height: 4px;
}

.c4 .MuiLinearProgress-colorPrimary {
  background-color: #1BBDCE;
}

.c4 .MuiLinearProgress-barColorPrimary {
  background-color: #102128;
}

@media (min-width:769px) {
  .c2 {
    font-size: 14px;
    line-height: 20px;
  }
}

@media (min-width:1024px) {
  .c2 {
    font-size: 16px;
    line-height: 24px;
  }
}

@media (min-width:769px) {
  .c2 {
    font-size: 22px;
    line-height: 24px;
    position: absolute;
    -webkit-transition: left 0.3s ease-in-out,-webkit-transform 0.3s ease-in-out;
    -webkit-transition: left 0.3s ease-in-out,transform 0.3s ease-in-out;
    transition: left 0.3s ease-in-out,transform 0.3s ease-in-out;
    left: 50%;
    -webkit-transform: translateX(-50%);
    -ms-transform: translateX(-50%);
    transform: translateX(-50%);
  }
}

@media (min-width:1024px) {
  .c2 {
    font-size: 30px;
    line-height: 30px;
  }
}

@media (min-width:769px) {
  .c3 {
    font-size: 14px;
    line-height: 20px;
  }
}

@media (min-width:1024px) {
  .c3 {
    font-size: 16px;
    line-height: 24px;
  }
}

@media (min-width:769px) {
  .c3 {
    font-size: 22px;
    line-height: 24px;
    position: absolute;
    -webkit-transition: left 0.3s ease-in-out,-webkit-transform 0.3s ease-in-out;
    -webkit-transition: left 0.3s ease-in-out,transform 0.3s ease-in-out;
    transition: left 0.3s ease-in-out,transform 0.3s ease-in-out;
    left: 50%;
    -webkit-transform: translateX(-50%);
    -ms-transform: translateX(-50%);
    transform: translateX(-50%);
  }
}

@media (min-width:1024px) {
  .c3 {
    font-size: 30px;
    line-height: 30px;
  }
}

@media (min-width:769px) {
  .c3 {
    left: 100%;
    -webkit-transform: none;
    -ms-transform: none;
    transform: none;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        Smarthome
      </div>
      <div
        class="c3"
      >
        Test
      </div>
    </div>
    <div
      class="c4"
    />
  </div>
</div>
`;
